src/hera/workflows/script.py

```diff
@@ -606,27 +617,83 @@ def _output_annotations_used(source: Callable) -> bool:
 
 FuncIns = ParamSpec("FuncIns")  # For input types of given func to script decorator
 FuncR = TypeVar("FuncR")  # For return type of given func to script decorator
+FuncRCov = TypeVar("FuncRCov", covariant=True)
```

Comment:
The type checker failed without it, complaining that we shouldn't be using an invariant TypeVar in the definition of `_ScriptDecoratedFunction` when we should be using a covariant one. But it is technically correct to do so: a `_ScriptDecoratedFunction[Foo, list[str]]` can safely be passed where a `_ScriptDecoratedFunction[Foo, Sequence[str]]` is required, for instance, because anything that returns a list is returning a sequence.